tests/shutdown/cleanup_before_shutdown.pm

```diff
@@ -21,6 +21,14 @@ use version_utils;
 
 sub run {
     select_console('root-console');
+    # Collect detailed logs to investigate shutdown issues and redirect them to serial console.
+    if (get_var('DEBUG_SHUTDOWN')) {
+        my $shutdown_debug_file         = '/usr/lib/systemd/system-shutdown/debug.sh';
+        my $shutdown_debug_file_content = "#!/bin/sh
```

Comment:
This script is small, but to be honest I would consider saving it in data directory and then curl it:
```perl
assert_script_run('curl ' . data_url("logs/debug.sh") -o <path>)
```
 like we do for other sh scripts there. LGTM other than that.

Reply:
A single execution script?

Reply:
We had a lot of issues with typing, so yes it's worth doing in cases of single usages. Here I've missed the point that we have variable, so it would be too much hustle.

Reply:
Changed to:

```perl
assert_script_run "echo -e '#!/bin/sh\\ndmesg > /dev/$serialdev' > /usr/lib/systemd/system-shutdown/debug.sh";
```